File: src/pages/login_page.py
"""Login page object (mobile + OTP). See context/ui-auth.md - OTP box selector unconfirmed."""

from playwright.sync_api import Page

from src.pages.base_page import BasePage


class LoginPage(BasePage):
    def __init__(self, page: Page, base_url: str) -> None:
        super().__init__(page, base_url)
        # --- Elements ---
        self.mobile_input = self.page.get_by_placeholder("Enter your mobile number")
        self.continue_btn = self.page.get_by_role("button", name="Continue")
        self.invalid_mobile_error = self.page.get_by_text("Invalid mobile number")
        self.otp_error_toast = self.page.get_by_text("Error in OTP")
        # OTP input boxes: exact selector unconfirmed (context/ui-auth.md Open Questions) -
        # OtpInput component, no data-testid found in source read this pass.

    def goto_login(self) -> None:
        self.goto("login")

    def fill_mobile_and_continue(self, mobile: str) -> None:
        self.mobile_input.fill(mobile)
        self.continue_btn.click()

    def fill_otp_and_continue(self, otp: str) -> None:
        # NOTE: OTP box selector unconfirmed - needs a live-inspected DOM check
        # before this method is usable. Placeholder raises until resolved.
        raise NotImplementedError(
            "OTP input selector not confirmed in context/ui-auth.md - "
            "resolve via live DOM inspection before use."
        )

File: src/pages/LoginPage.ts
import type { Locator, Page } from '@playwright/test';

/**
 * /login — mobile number + OTP authentication.
 * Selectors sourced from context/ui-auth.md (live DOM inspection, 2026-08-24).
 */
export class LoginPage {
  readonly page: Page;

  readonly mobileNumberInput: Locator;
  readonly termsCheckboxLabel: Locator;
  readonly continueButton: Locator;

  readonly otpInputs: Locator;
  readonly otpContinueButton: Locator;
  readonly resendOtpText: Locator;

  constructor(page: Page) {
    this.page = page;
    this.mobileNumberInput = page.getByRole('textbox', { name: 'Enter your mobile number' });
    this.termsCheckboxLabel = page.locator('label:has(input[data-testid="login_agreeTerms"])');
    this.continueButton = page.getByRole('button', { name: 'Continue' });

    this.otpInputs = page.getByTestId('otpVerify_otpInputs');
    this.otpContinueButton = page.getByRole('button', { name: 'Continue' });
    this.resendOtpText = page.getByText(/Request a new code in|Resend OTP/);
  }

  async goto(): Promise<void> {
    await this.page.goto('/login', { waitUntil: 'domcontentloaded' });
  }

  async enterMobileNumber(mobileNumber: string): Promise<void> {
    await this.mobileNumberInput.fill(mobileNumber);
  }

  async acceptTerms(): Promise<void> {
    await this.termsCheckboxLabel.click();
  }

  async submitMobileNumber(): Promise<void> {
    await this.continueButton.click();
  }

  async enterOtp(otp: string): Promise<void> {
    const digits = otp.split('');
    for (let i = 0; i < digits.length; i += 1) {
      await this.otpInputs.nth(i).fill(digits[i]);
    }
  }

  async submitOtp(): Promise<void> {
    await this.otpContinueButton.click();
  }

  /** Full mobile+OTP login in one call — used by global-setup and the login flow's own test cases. */
  async login(mobileNumber: string, otp: string): Promise<void> {
    await this.goto();
    await this.enterMobileNumber(mobileNumber);
    await this.acceptTerms();
    await this.submitMobileNumber();
    await this.otpInputs.first().waitFor({ state: 'visible' });
    await this.enterOtp(otp);
    await this.submitOtp();
  }
}
